velox/common/caching/SsdFile.cpp

```diff
@@ -859,35 +842,25 @@ void SsdFile::initializeCheckpoint() {
     return;
   }
 
-  bool hasCheckpoint = true;
-  std::unique_ptr<common::FileInputStream> checkpointInputStream;
   filesystems::FileOptions writeFileOptions;
   writeFileOptions.shouldThrowOnFileAlreadyExists = false;
 
   const auto checkpointPath = getCheckpointFilePath();
   try {
     checkpointWriteFile_ =
         fs_->openFileForWrite(checkpointPath, writeFileOptions);
-
-    auto checkpointReadFile = fs_->openFileForRead(checkpointPath);
-    checkpointInputStream = std::make_unique<common::FileInputStream>(
-        std::move(checkpointReadFile),
-        1 << 20,
-        memory::memoryManager()->cachePool());
-  } catch (std::exception& e) {
-    hasCheckpoint = false;
-    ++stats_.openCheckpointErrors;
-    VELOX_SSD_CACHE_LOG(WARNING) << fmt::format(
-        "Error openning checkpoint file {}: Starting shard {} without checkpoint, with checksum write {}, read verification {}, checkpoint file {}",
-        e.what(),
-        shardId_,
-        checksumEnabled_ ? "enabled" : "disabled",
-        checksumReadVerificationEnabled_ ? "enabled" : "disabled",
-        checkpointPath);
+  } catch (const std::exception& e) {
+    ++stats_.writeCheckpointErrors;
+    VELOX_SSD_CACHE_LOG(ERROR) << fmt::format(
+        "Could not initilize checkpoint file {} for writing: {}: ",
+        checkpointPath,
+        e.what());
   }
 
   const auto logPath = getEvictLogFilePath();
   try {
+    filesystems::FileOptions writeFileOptions;
```

Comment:
Can you make a static or anonymous function for this? Thanks!

Reply:
Realize we can reuse the one created above.